server/controllers/wip/questionsController.ts

```diff
@@ -1,3 +1,111 @@
+import { FormWizard } from 'hmpo-form-wizard'
+import type express from 'express'
 import { BaseController } from '../index'
+import { ReportWithDetails } from '../../data/incidentReportingApi'
+import format from '../../utils/format'
+import { getIncidentTypeConfiguration } from '../../reportConfiguration/types'
+import {
+  AnswerConfiguration,
+  IncidentTypeConfiguration,
+  QuestionConfiguration,
+} from '../../data/incidentTypeConfiguration/types'
+import logger from '../../../logger'
 
-export default class QuestionsController extends BaseController {}
+export default class QuestionsController extends BaseController<FormWizard.MultiValues> {
+  getBackLink(_req: FormWizard.Request, _res: express.Response): string {
+    return '/reports/'
+  }
+
+  middlewareLocals(): void {
+    this.use(this.lookupReport)
+    super.middlewareLocals()
+  }
+
+  async lookupReport(req: FormWizard.Request, res: express.Response, next: express.NextFunction): Promise<void> {
+    const { incidentReportingApi } = res.locals.apis
+    const reportId = req.params.id
+
+    res.locals.report = await incidentReportingApi.getReportWithDetailsById(reportId)
+    res.locals.reportConfig = await getIncidentTypeConfiguration(res.locals.report.type)
+    next()
+  }
+
+  getValues(
+    req: FormWizard.Request<FormWizard.MultiValues, string>,
+    res: express.Response,
+    callback: FormWizard.Callback<FormWizard.MultiValues>,
+  ) {
+    return super.getValues(req, res, async (err, values) => {
+      if (err) {
+        return callback(err)
```

Comment:
this might work perfectly in practice now, but do be aware that in _general_ it's not a great idea to return the return value of a callback function. that could be an unrelated promise if the callback is async.

safer to:
```typescript
callback()
return
```

this point comes from being stung in the past

Reply:
I'll change 👍

Reply:
59d34e39f9d961185426b4d28537fba24b1aeca7